File: a-book-organizer.js
const books = [
  {
    title: "The Hobbit",
    authorName: "J.R.R. Tolkien",
    releaseYear: 1937
  },
  {
    title: "1984",
    authorName: "George Orwell",
    releaseYear: 1949
  },
  {
    title: "To Kill a Mockingbird",
    authorName: "Harper Lee",
    releaseYear: 1960
  }
];

function sortByYear(book1, book2) {
  if (book1.releaseYear < book2.releaseYear) {
    return -1;
  } else if (book1.releaseYear > book2.releaseYear) {
    return 1;
  } else {
    return 0;
  }
}

const filteredBooks = books.filter(book => book.releaseYear <= 1950);

filteredBooks.sort(sortByYear);

File: a-book-organizer.py
books = [
    {"title": "The Hobbit", "authorName": "J.R.R. Tolkien", "releaseYear": 1937},
    {"title": "1984", "authorName": "George Orwell", "releaseYear": 1949},
    {"title": "To Kill a Mockingbird", "authorName": "Harper Lee", "releaseYear": 1960},
]


def sort_by_year(book):
    return book["releaseYear"]


filtered_books = [book for book in books if book["releaseYear"] <= 1950]

filtered_books.sort(key=sort_by_year)

print(filtered_books)